codalab/rest/bundles.py

```diff
@@ -426,6 +425,67 @@ def _fetch_locations():
     return dict(data=uuids_to_locations)
 
 
+@get('/bundle_stores/', apply=AuthenticatedProtectedPlugin())
+def _fetch_bundle_stores():
+    """
+    Fetch the bundle stores available to the user.
+
+    Returns a dictionary in which the keys are the bundle store uuids, and the values
+    are tuples with the owner_id, name, and url.
+    """
+    return local.model.get_bundle_stores(request.user)
+
+
+@post('/bundle_stores/', apply=AuthenticatedProtectedPlugin())
+def _add_bundle_store():
+    """
+    Add a bundle store that the user can access.
+    """
+    return local.model.create_bundle_store(
+        request.user,
+        request.forms.get('name'), 
```

Comment:
Don't use `request.forms` because that would use form bodies. We use JSON bodies everywhere else. See https://stackoverflow.com/questions/23118249/whats-the-difference-between-request-payload-vs-form-data-as-seen-in-chrome for more info on the difference.

Let's use `request.json` and create a marshmallow schema to validate input, similar to `BundlePermissionSchema` (see https://github.com/codalab/codalab-worksheets/blob/6cf9a3159c3b332d173c521ca516ed69247fe2f6/codalab/rest/bundles.py#L406 for an example)

Reply:
Also do this for the update bundle store endpoint